docker/Dockerfile

```diff
@@ -0,0 +1,107 @@
+# build stage for compiling odamex
+FROM ubuntu:20.04 AS build
+WORKDIR /build
+ENV DEBIAN_FRONTEND=noninteractive
+RUN true \
+    && apt-get update -qq \
+    && apt-get install -y --no-install-recommends tzdata\
+    && apt-get install -y -qq --no-install-recommends \
+        ca-certificates \
+        git \
+        gcc g++ \
+        cmake \
+        make \
+        deutex \
+        zlib1g-dev \
+        libgl1-mesa-dev \
+        libssl-dev \
+        libcurl4-openssl-dev libcurlpp-dev \
+        patch \
+        > /dev/null
+
+# So we can use bash arrays (default /bin/sh doesn't support this)
+SHELL ["/bin/bash","-c"]
+
+ARG REPO_URL
+ARG REPO_TAG
+
+# Clone the repository
+RUN true \
+    && test -n "$REPO_URL" && test -n "$REPO_TAG" \
+    && git clone --depth 1 --branch "$REPO_TAG" "$REPO_URL" odamex
+
+WORKDIR /build/odamex
+
+# Apply Manual Patches (make sure they are UTF-8 encoded)
+RUN mkdir /patches
+COPY docker-files/patches /patches
+RUN true \
+    && shopt -s nullglob \
+    && for p in /patches/*.patch; do patch -p1 < $p; done
+
+# Update submodules
+RUN true \
+    && git submodule init \
+    && git submodule update
```

Comment:
This should probably be submodule update -r

Reply:
Done